packages/ckeditor5-engine/src/view/renderer.ts

```diff
@@ -667,6 +680,12 @@ export default class Renderer extends Observable {
 
 		const diff = this._diffNodeLists( actualDomChildren, expectedDomChildren );
 
+		// We need to check whether there are some text nodes to update data to make sure that replacing such nodes
```

Comment:
I couldn't understand this explanation.

I'd start from what's different between the flow for Android and other browsers. Explain first the context.

Then try to explain why it's here and how it changes the behavior later on. It's hard for me e.g. to understand what's the difference between `diff` initially and after passing it via `actions`. And then I'd need to understand why that difference was needed.

We may also need to clarify why we don't do `findReplaceActions()` on other browsers. Especially if the reason was "we don't want to change the behavior on other browsers for safety, but maybe one day cause it seems to make sense".